src/test/java/core/basesyntax/service/impl/WriteToCsvFileServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+                    Operation.SUPPLY, new SupplyOperationHandlerImpl());
+
+    @BeforeAll
+    static void beforeAll() {
+        dataConverter = new DataConvertServiceImpl();
+        dataProcessor = new DataProcessServiceImpl(operationPicker);
+    }
+
+    @BeforeEach
+    void setUp() {
+        reportCreator = new CreateReportServiceImpl();
+        csvReader = new ReadFromCsvFileServiceImpl();
+        csvWriter = new WriteToCsvFileServiceImpl(REPORT_FILENAME);
+        builder = new StringBuilder();
+    }
+
+    @Test
+    void csvWriter_twoFruits_Ok() {
+        dataProcessor.processFruits(
+                dataConverter.convert(csvReader.readFile(FIRST_FILENAME)));
+
+        String expected = reportCreator.createReport();
+        csvWriter.write(expected);
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(REPORT_FILENAME))) {
+            String line = reader.readLine();
+
+            while (line != null) {
+                builder.append(line)
+                        .append(System.lineSeparator());
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot find file: " + REPORT_FILENAME);
+        }
+        String actual = builder.toString().trim();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void csvWriter_threeFruits_Ok() {
+        dataProcessor.processFruits(
+                dataConverter.convert(csvReader.readFile(SECOND_FILENAME)));
+
+        String expected = reportCreator.createReport();
+        csvWriter.write(expected);
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(REPORT_FILENAME))) {
+            String line = reader.readLine();
+
+            while (line != null) {
+                builder.append(line)
+                        .append(System.lineSeparator());
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot find file: " + REPORT_FILENAME);
+        }
+        String actual = builder.toString().trim();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void csvWriter_fourFruits_Ok() {
+        dataProcessor.processFruits(
+                dataConverter.convert(csvReader.readFile(THIRD_FILENAME)));
+
+        String expected = reportCreator.createReport();
+        csvWriter.write(expected);
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(REPORT_FILENAME))) {
+            String line = reader.readLine();
+
+            while (line != null) {
+                builder.append(line)
+                        .append(System.lineSeparator());
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot find file: " + REPORT_FILENAME);
+        }
+        String actual = builder.toString().trim();
+
+        assertEquals(expected, actual);
+    }
```

Comment:
let's try to avoid code duplication here

Reply:
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/46851188/417493e1-947e-4def-a8a6-18766ba3bf2d)